api/v1alpha2/jenkins_webhook.go

```diff
@@ -41,21 +48,115 @@ var _ webhook.Validator = &Jenkins{}
 
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type
 func (in *Jenkins) ValidateCreate() error {
-	jenkinslog.Info("validate create", "name", in.Name)
+	if in.Spec.ValidateSecurityWarnings {
+		jenkinslog.Info("validate create", "name", in.Name)
+		return Validate(*in)
+	}
 
-	// TODO(user): fill in your validation logic upon object creation.
 	return nil
 }
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (in *Jenkins) ValidateUpdate(old runtime.Object) error {
-	jenkinslog.Info("validate update", "name", in.Name)
+	if in.Spec.ValidateSecurityWarnings {
+		jenkinslog.Info("validate update", "name", in.Name)
+		return Validate(*in)
+	}
 
-	// TODO(user): fill in your validation logic upon object update.
 	return nil
 }
 
 func (in *Jenkins) ValidateDelete() error {
-	// TODO(user): fill in your validation logic upon object deletion.
+	return nil
+}
+
+type Warnings struct {
+	Warnings []Warning `json:"securityWarnings"`
+}
+
+type Warning struct {
+	Versions []Version `json:"versions"`
+	ID       string    `json:"id"`
+	Message  string    `json:"message"`
+	URL      string    `json:"url"`
+	Active   bool      `json:"active"`
+}
+type Version struct {
+	FirstVersion string `json:"firstVersion"`
+	LastVersion  string `json:"lastVersion"`
+}
+
+const APIURL string = "https://plugins.jenkins.io/api/plugin/"
+
+func checkSecurityWarnings(pluginName string, pluginversion string) bool {
+	jenkinslog.Info("checking security warnings", "plugin: ", pluginName)
+	pluginURL := APIURL + pluginName
+	client := &http.Client{}
+	request, err := http.NewRequest("GET", pluginURL, nil)
+	if err != nil {
+		log.Fatal("unable to create a new request", err.Error())
```

Comment:
Is `log.Fatal` the right choice here? It will call `os.Exit(1)` which exits the whole process. That might not be the wise thing to do.

Reply:
Yeah you are right it should return an error failing creating a new object but the operator should be able to run.
Thanks for the insightful comments, I will work on it.

Reply:
No problem! Thanks for contributing! 🙂